pkg/lib/kitfile/local-storage.go

```diff
@@ -89,64 +90,81 @@ func saveConfig(ctx context.Context, localRepo local.LocalRepo, kitfile *artifac
 	return desc, nil
 }
 
+// SaveKitfileLayers saves the kitfile layers with optional compression and returns a list of descriptors.
 func saveKitfileLayers(ctx context.Context, localRepo local.LocalRepo, kitfile *artifact.KitFile, ignore filesystem.IgnorePaths, compression string) ([]ocispec.Descriptor, error) {
 	var layers []ocispec.Descriptor
+	var mu sync.Mutex
+	var wg sync.WaitGroup
+	errChan := make(chan error, len(kitfile.Model.Parts)+len(kitfile.Code)+len(kitfile.DataSets)+len(kitfile.Docs))
```

Comment:
I think `kitfile.Model` may be nil here, meaning the `kitfile.Model.Parts` access has to have a guard.

Reply:
added a nil pointer check!